src/snap.c

```diff
@@ -11,6 +11,85 @@
 #include "snap.h"
 #include "view.h"
 
+static struct {
```

Comment:
Suggest adding a comment here to describe why we have the cache. The text in the commit message would be good.